GVRf/Framework/jni/objects/components/texture_capturer.cpp

```diff
@@ -164,8 +167,10 @@ glm::mat4 TextureCapturer::getMvpMatrix(float half_width, float half_height) {
 }
 
 void TextureCapturer::callback(int msg, char *info) {
-    Java_org_gearvrf_NativeTextureCapturer_callbackFromNative(
-            mJNIEnv, mCapturerObject, msg, info);
+    if (nullptr != mCapturerObject) {
```

Comment:
I had no idea of the assumptions in this code. It looked to me like the native side wanted a guarantee that the java object is alive at the time of the callback. A guarantee a weak reference cannot give. What I made felt like the safer fix. And after all, if nobody listens why send events?